foxglove_bridge_base/include/foxglove_bridge/websocket_server.hpp

```diff
@@ -521,20 +640,88 @@ inline void Server<ServerConfiguration>::handleMessage(ConnHandle hdl, MessagePt
           _unsubscribeHandler(chanId, hdl);
         }
       }
+    } break;
+    case Integer("advertise"): {
+      for (const auto& chan : payload.at("channels")) {
+        ClientChannelId channelId = chan.at("id");
+        if (_clientChannels.find(channelId) != _clientChannels.end()) {
+          sendStatus(hdl, StatusLevel::ERROR,
+                     "Channel " + std::to_string(channelId) + " was already advertised");
+          continue;
+        }
+        ClientAdvertisement advertisement{};
+        advertisement.channelId = channelId;
+        advertisement.topic = chan.at("channelId").get<std::string>();
+        advertisement.encoding = chan.at("encoding").get<std::string>();
+        advertisement.schemaName = chan.at("schemaName").get<std::string>();
+        _clientChannels.emplace(channelId, std::move(advertisement));
```

Comment:
```suggestion
        _clientChannels.emplace(channelId, advertisement);
```

Because `advertisement` is passed as a const ref to `_clientAdvertiseHandler` below